### ISSUE TITLE
libEGL warning: DRI2: xcb_connect failed -- version 2.0.0

### ISSUE THREAD

**Issue by aleksejleonov** [klokantech/tileserver-gl#185]
When trying to start klokantech/tileserver-gl (v2.0.0) docker container (using Docker for Mac) got next error:

```bash
...
Run with --verbose to see the config file here.
libEGL warning: DRI2: xcb_connect failed
terminate called after throwing an instance of 'std::runtime_error'
  what():  eglInitialize() failed.

Aborted
```

Full log:
```bash
$ docker run -it -v $(pwd):/data -p 8080:80 klokantech/tileserver-gl
Unable to find image 'klokantech/tileserver-gl:latest' locally
latest: Pulling from klokantech/tileserver-gl
c75480ad9aaf: Already exists 
1fc532d0fa06: Pull complete 
f8d5f60b11a6: Pull complete 
fbabebf70261: Pull complete 
a7adb05b846e: Pull complete 
d0eff0fc189b: Pull complete 
Digest: sha256:e3b58b5a16e099f4c03f2b40d3e9b4a70a859e3fb6284df6fdb83ad662a20602
Status: Downloaded newer image for klokantech/tileserver-gl:latest
Starting tileserver-gl v2.0.0
No MBTiles specified, using planet.mbtiles
Automatically creating config file for planet.mbtiles
Run with --verbose to see the config file here.
libEGL warning: DRI2: xcb_connect failed
terminate called after throwing an instance of 'std::runtime_error'
  what():  eglInitialize() failed.

Aborted
```

Previous version (1.7.0) works fine:
```bash
$ docker run --rm -it -v $(pwd):/data -p 8080:8080 klokantech/tileserver-gl:v1.7.0
Unable to find image 'klokantech/tileserver-gl:v1.7.0' locally
v1.7.0: Pulling from klokantech/tileserver-gl
d31335c0dd82: Pull complete 
9fc2484e8870: Pull complete 
cf6bed6de800: Pull complete 
cab4f94122a7: Pull complete 
2817fc3d734d: Pull complete 
4e49ab86bb83: Pull complete 
Digest: sha256:e4fa28aed86e47361831a65cab3c7c658db69dbbcd6461e623eb75697a020e83
Status: Downloaded newer image for klokantech/tileserver-gl:v1.7.0
Starting tileserver-gl v1.7.0
No MBTiles specified, using planet.mbtiles
Automatically creating config file for planet.mbtiles
Run with --verbose to see the config file here.
Starting server
Listening at http://[::]:80/
Startup complete
``` 

Docker for Mac 17.06.0-ce-mac18 (stable channel), MacBook Pro 15" Mid 2014, macOS 10.12.5 (16F73).

**Comment by petrsloup** [comment]
I tried starting up the container several times and I got the error only once.
There was a possible race condition in xvfb startup and it was probably not yet initialized properly.

@aleksejleonov Can you please try pulling the latest image and test if the problem is now solved on your system?

**Comment by aleksejleonov** [comment]
@petrsloup latest image work like a charm.
Thank you 👍

**Comment by snoopckuu** [comment]
I am having the same issue with the latest image 306cfd331996

tile_1          | libEGL warning: DRI3: xcb_connect failed
tile_1          | libEGL warning: DRI2: xcb_connect failed
tile_1          | libEGL warning: DRI2: xcb_connect failed
tile_1          | terminate called after throwing an instance of 'std::runtime_error'
tile_1          |   what():  eglInitialize() failed.
tile_1          | 
tile_1          | /usr/src/app/run.sh: line 8:    11 Aborted                 (core dumped) node /usr/src/app/ -p 80 "$@"

**Comment by simoncas** [comment]
I encounter the same problem than @snoopckuu & @aleksejleonov 
`
map_1      | libEGL warning: DRI3: xcb_connect failed
map_1        | libEGL warning: DRI2: xcb_connect failed
map_1        | libEGL warning: DRI2: xcb_connect failed
map_1        | terminate called after throwing an instance of 'std::runtime_error'
map_1        |   what():  eglInitialize() failed.
map_1        | 
map_1        | /usr/src/app/run.sh: line 8:     9 Aborted                 (core dumped) node /usr/src/app/ -p 80 "$@"
`

**Comment by sclausen** [comment]
I found out, it just occurs, if I run the container more than once. If I delete the container and the image, the error never occurs.

**Comment by edvinbrobeck** [comment]
Any updates regarding this?

**Comment by petrsloup** [comment]
Have you tried adding `--rm` to the `docker run` command (https://docs.docker.com/engine/reference/run/#clean-up-rm) to ensure everything is cleaned-up after the process exits?

Looks like this might help if the problem occurs only after running more than once...

**Comment by sclausen** [comment]
@petrsloup Any suggestion using something like `--rm` with docker-compose?

**Comment by dadwin** [comment]
@sclausen I'm using `docker-compose down` and after that I don't get "xcb_connect failed".

**Comment by petrsloup** [comment]
Are you still having issues after the https://github.com/klokantech/tileserver-gl/pull/242 was merged? (Try version 2.3.0).
